parcels/xgrid.py

```diff
@@ -271,7 +271,10 @@ def _gtype(self):
     def search(self, z, y, x, ei=None):
         ds = self.xgcm_grid._ds
 
-        zi, zeta = _search_1d_array(ds.depth.values, z)
+        if hasattr(ds, "depth"):
```

Comment:
I think this would be better than a `hasattr` check
```suggestion
        if "Z" in self.grid.axes:
```